### ISSUE TITLE
Add killfile support

### ISSUE THREAD

**Issue by pseudomonas** [dreamwidth/dw-free#1552]
This is a placeholder for a copy of bugzilla bug № 35, the text and specifications of which were unfortunately in a bugzilla attachment and are no longer extant.

The essence of the proposal is that a user should be able to opt-out of seeing activity (such as comments & notifications) from another user.

**Comment by azurelunatic** [comment]
Found a copy:
http://dreamwidth-meta.dreamwidth.org/887.html?thread=21879#t21879

**Comment by azurelunatic** [comment]
Add killfile support to the service, wherein people can designate other
people as "do not show me this user's content". This project includes:

* Two levels of killfile:

1). "collapse all content by this user, but still show me that the content
exists", where any community entries, comments, etc by that user will just
show as "[Post by username]" or "[Comment by username]" but can be clicked
on to display the content if the user wishes, and

2). "never show me evidence that this user exists at all", where any
community entries, comments, etc by that user will simply not be displayed.
Level 2 killfile should also remove that username from profile pages you're
viewing, although the profile should always include a note that people you
have killfiled are members of the comm/trusted by the user/etc, along with
a link that, if clicked, will display the usernames of which killfiled
names have been removed. (This is to avoid someone not realizing that their
mortal enemy is on a third party's trust list and may be viewing the same
locked post, etc.)

* A user-facing killfile management page, where you can see everyone you've
killfiled and what their *plonk* level is. (A *plonk*, for non-Usenetters,
is the sound someone makes when they hit the bottom of your killfile.)

* Admin diagnostic tools: people with the appropriate privs can see someone
else's killfile settings, to verify if display problems are
killfile-related.

**Comment by anall** [comment]
I remember the official spec being more RFC-ish with lots of "MUST" and "SHOULD"s

**Comment by azurelunatic** [comment]
Some thoughts from dreamwidth-dev: 

* This might be implemented console-first, and adding some of the UI might be done separately, especially if it's more eff0rt-minor
* Controls to set and remove killfile should be in the interaction menu on the profile
* It should show the current status on the interaction menu on the profile, like circle membership does 
* Level 2 should probably *not* be on the popup
* Some suggestions for what to call it include Mute, Hide
* Must not show to the muted user that they are muted, even though they may be able to make an educated guess
* Level 2 might affect the size of the reading page, if entries are pulled first and hidden second. Performance people, would it be okay to check for Level 2 muted entries before finishing building the reading page and pull extra to fill the gaps? If this would cause Woe (especially at ?skip=300 where you'd have to know what was in the previous 280) then this could probably be solved with documentation.

**Comment by rahaeli** [comment]
I went looking for it but I don't have it saved locally, sigh.

**Comment by pseudomonas** [comment]
With level 2, if a muted user makes a comment, should replies to that comment be shown as top-level? Or do we put in a placeholder *iff* there's replies (exactly as for deleted and screened comments)

**Comment by rahaeli** [comment]
Thinking about this a bit, along with the associated questions, and I think we can skip implementing level-2-killfile -- I don't see where it gives us *much* more than hide-all-comments/entries-from-this-user, and it produces way too many questions that will complicate the hell out of implementation. A simple "collapse comments/entries from this person by default" should suffice for a first pass.

**Comment by pseudomonas** [comment]
That should make things *a lot* easier to implement.

**Comment by pseudomonas** [comment]
claim

**Comment by zorkian** [comment]
+1, I think the first level makes sense and I'm OK with. The second level I have some moral quandary about, so, if we're going to do that I'd like to be involved in some small discussion.

But definitely the first level is great and should be awesome.

**Comment by rahaeli** [comment]
Yeah, I think that "nuke it from orbit" level killfiling might still have its place at some point, but it's more complex and will require a lot more careful thought first. "Hide everything from this person" is a lot simpler in both coding/design and in possible-social-repercussions.

**Comment by ninetyd** [comment]
Recent request from a user who wanted to exclude a community poster from their Reading list: https://www.dreamwidth.org/support/see_request?id=35675
